subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -98,6 +104,7 @@ public SigningExtension(Project project) {
         this.configuration = getDefaultConfiguration();
         this.signatureTypes = createSignatureTypeProvider();
         this.signatories = createSignatoryProvider();
+        this.signTasks = createSignTasksMap();
```

Comment:
I would simply instantiate the Map here. All the other methods have logic in them, which is why there is indirection.